developing/vcs/README.md

```diff
@@ -4,25 +4,50 @@
 
 Source control has less to do with the VCS system you use, and more to do with the stages code goes through on the journey from inside a developer's head, tested and merged into the main line, deployed to production, and thence in time to being identified as that accursed legacy system that is causing all the problems.
 
-## Repo maintenance
+## Repo maintenance and hygiene
 
 * Avoid including files in source control that are specific to your development machine or process.
-* Ignore files that are specific to your machine or process in your `~/.gitignore_global` as opposed to your repo's `.gitignore`.
-* Delete local and remote feature branches after merging.
-* Perform work in a feature branch.
-* Rebase frequently to incorporate upstream changes.
-* Use a pull request for code reviews.
+* Ignore files that are specific to your machine or process in your
+`~/.gitignore_global` as opposed to your repo's `.gitignore`.
+* Delete local and remote feature branches after merging. This can be automated
+on the GitHub side with [Probot](https://probot.github.io/apps/delete-merged-branch/).
+Locally, you can add an alias to your `~/.gitconfig` file. Example:
+
+  ```
+  [alias]
+    fresh = "!git co master && git pull && git branch --merged | grep -v '\\*' | xargs -n 1 git branch -d"
```

Comment:
There is `git fetch --prune`

Reply:
Which one is preferable?

Reply:
Exposing my thinking for sanity checking what I believe these commands are doing:

* `fetch --prune` deletes local branches that are no longer present on the tracked remote.
* `fresh` (as defined here) looks like it is a bit more aggressive: it pulls to master and then deletes any local branches that are reachable from (have been merged to) master (excluding master itself), regardless of whether they have been deleted on the remote.

That might mean that a branch that has been deleted on the remote without being merged to master would remain in the local repo after `fresh`, but I haven't tested this.

@monfresh Am I in the neighborhood?

Which one is preferable, I'm not sure. We are recommending automating deleting merged branches, but `fresh` would get rid of them locally if that's not the case.

Reply:
@monfresh Doesn't `fresh` delete integration branches if you happen to run it straight after integration is merged to master?

Reply:
Are you saying that running `git fetch --prune` by itself should delete local branches that have been deleted from the remote? It's not working for me. When I run it, I see something like this:
```
 - [deleted]         (none)     -> origin/mb-squash-merge
```
but it doesn't actually delete my local branch. What am I missing?

Also, ideally, local branches would be deleted even if the branch hasn't yet been deleted from the remote. As long as it is merged, we should be able to delete the local branch.

Reply:
> Are you saying that running git fetch --prune by itself should delete local branches that have been deleted from the remote?

That is what I was saying, but I tried it again and I was wrong. From `git help fetch`:

>        -p, --prune
>           Before fetching, remove any remote-tracking references that no longer exist on the remote

So `--prune` removes the *tracking refererence* (what shows up in `git branch -v as something like `remotes/origin/branchname`) -- probably a good thing, but it does not delete the local branch at all. That still requires `branch -d`. 

I suppose an integration branch deleted by `fresh` could be retrieved with `fetch origin/integration`. So even in the worst case this shouldn't actually destroy any work.